### PR TITLE
Add "What is Azure Key Vault" SEO Page

### PR DESCRIPTION
**Pull request by toriancrane**
## Description

This PR stages the content for the "What is Azure Key Vault" SEO page.

It additionally:

- Adds a '?' to the page titles of the Docker Configs/Secrets pages
- Corrects the headers on the same pages
- Makes small edits to the content of the same pages

### CODE REVIEW

File: themes/default/content/what-is/what-is-azure-key-vault.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+Microsoft Azure is a leader in cloud computing, transforming the way organizations manage their digital infrastructure. An important aspect of Azure’s security framework is the management of sensitive data, commonly known as "[secrets](/what-is/what-is-secrets-management/)". Azure Key Vault is a service designed for the secure handling of these secrets, offering tools for storing, accessing, and managing confidential information in the cloud.
+
+## What is Azure Key Vault?
+
+Azure Key Vault is a cloud service for managing, retrieving, and storing sensitive information such as database credentials, API keys, and other secrets. It helps in securing access to applications, services, and IT resources without hard-coding sensitive information in plain text.
+
+### Key features
+
+- **Secure Storage**: Azure Key Vault offers highly secure storage for secrets, keys, and certificates, encrypted both in transit and at rest.
```
Comment (desteves) [inline themes/default/content/what-is/what-is-azure-key-vault.md]: For anything that's bulleted, we're using sentence casing

File: themes/default/content/what-is/what-is-azure-key-vault.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+- **Seamless Integration**: It integrates effortlessly with other Azure services and applications, enhancing overall security architecture.
+- **Access Control**: Azure Key Vault allows fine-grained access control with Microsoft Entra and access policies.
+
+## Creating Azure Key Vault secrets
+
+Azure Key Vault secrets can be created via the Azure CLI. Before creating secrets in Azure, you must first make sure you have the [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) installed. Once you have installed the Azure CLI, run the `az login` command to [authenticate to your Azure account](https://learn.microsoft.com/en-us/cli/azure/authenticate-azure-cli).
+
+```bash
+$ az login
+To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code LZURMHUR9 to authenticate.
```
Comment (desteves) [inline themes/default/content/what-is/what-is-azure-key-vault.md:29]: Maybe omit the output of az login so it's cleaner? The code will also differ

File: themes/default/content/what-is/what-is-azure-key-vault.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    "managed": null,
+    "name": "MySecretName",
+    "tags": {
+      "file-encoding": "utf-8"
+    }
+  }
+]
+```
+
+{{< notes type="info" >}}
```
Comment (desteves) [inline themes/default/content/what-is/what-is-azure-key-vault.md:143]: Nice note!

**Review by desteves — APPROVED** [review]
Minor comments. LGTM

File: themes/default/content/what-is/what-are-docker-configs.md [hunk 4]
```diff
@@ -4,24 +4,24 @@ meta_desc: |
...
+Docker, a leading platform in containerization technology, has revolutionized how applications are developed, shipped, and deployed. An essential facet of this ecosystem is the effective management of configuration data. [Docker Configs](https://docs.docker.com/engine/swarm/configs/) is a feature specially crafted to handle non-sensitive configuration information within Docker environments.
 
-### What are Docker Configs?
+## What are Docker Configs?
 
 Docker Configs are a resource in Docker for storing non-sensitive information such as configuration files, separate from a service's image or running containers within [Docker Swarm](https://docs.docker.com/engine/swarm/) environments. This enables keeping Docker images as generic as possible without the need for bind-mounting configuration files into containers or using environment variables. Unlike [Docker Secrets](/what-is/what-are-docker-secrets), Docker Configs are not encrypted at rest and are directly mounted into the container's filesystem.
 
-#### Key Features
+### Key Features
```
Comment (scottslowe) [inline themes/default/content/what-is/what-are-docker-configs.md]: Our style guide calls for sentence casing for headings

File: themes/default/content/what-is/what-are-docker-configs.md [hunk 5]
```diff
@@ -4,24 +4,24 @@ meta_desc: |
...
 
 - **Separation of Configuration from Code**: Docker Configs allow you to store configuration files outside of your Docker images, leading to more generic and reusable images.
```
Comment (scottslowe) [inline themes/default/content/what-is/what-are-docker-configs.md]: I'd also recommend using sentence casing on these initial bullet point headings

**Review by scottslowe — CHANGES_REQUESTED** [review]
Just some issues with sentence casing in the headings (both the standalone headings and the bulleted list headings); otherwise, looks good.

**Pull request merged**